lib/chewy/search/parameters.rb

```diff
@@ -165,6 +182,116 @@ def render_query
           {query: {bool: filter}}
         end
       end
+
+    private
+
+      # Smartly wraps a query in a bool must unless it is already correctly structured.
+      # This method helps maintain logical grouping and avoid unnecessary nesting in queries.
+      #
+      # @param [Hash, Array, Nil] query The query to wrap.
+      # @return [Hash, Array, Nil] The wrapped or original query.
+      #
+      # Example:
+      #   input: { term: { status: 'active' } }
+      #   output: { bool: { must: [{ term: { status: 'active' } }] } }
+      #
+      #   input: { bool: { must: [{ term: { status: 'active' } }] } }
+      #   output: { bool: { must: [{ term: { status: 'active' } }] } }
+      def smart_wrap_in_bool_must(query = nil)
```

Comment:
Custom merge strategy for avoiding unnecessary nesting

We had added this custom change already in our forked version

Reply:
I believe I already removed this before :)

Reply:
Removed ?